sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -356,15 +367,11 @@ case class MaxOf(left: Expression, right: Expression) extends BinaryArithmetic {
           ${ev.isNull} = ${eval1.isNull};
           ${ev.primitive} = ${eval1.primitive};
         } else {
-          if (${eval1.primitive} > ${eval2.primitive}) {
-            ${ev.primitive} = ${eval1.primitive};
-          } else {
-            ${ev.primitive} = ${eval2.primitive};
-          }
+          $result
         }
       """
     } else {
-      super.genCode(ctx, ev)
+      ctx.defaultCodeGen(ev, this)
```

Comment:
The origin code is not right as `super` here refers to `BinaryArithmetic`, not `Expression` as we wanted.